usr/src/lib/nsswitch/files/common/getnetgrent.c

```diff
@@ -0,0 +1,879 @@
+		for (cur = ngt->hash_heads[i];  cur != 0; /* cstyle */) {
+			next = cur->hash_chain;
+			free(cur);
+			cur = next;
+		}
+	}
+	/* Don't bother zeroing pointers;  must do init if we want to reuse */
+}
+
+typedef const char *ccp;
+
+static nss_status_t
+top_down(struct files_backend *be, const char **groups, int ngroups,
+    int (*func)(ccp triple[3], void *iter_args, nss_status_t *return_val),
+    void *iter_args)
+{
+	struct netgrtab		*ngt;
+	/* netgrtab goes on the heap, not the stack, because it's large and */
+	/* stacks may not be all that big in multi-threaded programs. */
+
+	const char		*group;
+	int			nfound;
+	int			done;
+	nss_status_t		result;
+
+	if ((ngt = (struct netgrtab *)malloc(sizeof (*ngt))) == 0) {
+		return (NSS_UNAVAIL);
+	}
+	ngt_init(ngt);
+
+	while (ngroups > 0) {
+		ngt_insert(ngt, *groups, strlen(*groups));
+		groups++;
+		ngroups--;
+	}
+
+	done	= 0;	/* Set to 1 to indicate that we cut the iteration  */
+			/*   short (and 'result' holds the return value)   */
+	nfound	= 0;	/* Number of successful netgroup getbyname calls   */
+
+	while (!done && (group = ngt_next(ngt)) != 0) {
+		char		*val = NULL;
+		char		*p;
+
+		result = netgr_get_members(be, group, &val);
+		if (result != NSS_SUCCESS) {
+			/*LINTED E_NOP_IF_STMT*/
+			if (result == NSS_NOTFOUND) {
+				if (__nss_files_netgr_error != NULL)
+					__nss_files_netgr_error(
+			    "files netgroup lookup: %s doesn't exist",
+					    group);
+			} else {
+				if (__nss_files_netgr_error != NULL)
+					__nss_files_netgr_error(
+			"files netgroup lookup: getbyname returned [%s]",
+					    strerror(errno));
+				done = 1;	/* Give up, return result */
+			}
+			/* Don't need to clean up anything */
+			continue;
+		}
+
+		if (__nss_files_netgr_debug != NULL)
+			__nss_files_netgr_debug(
+			    "ngt_top: ngt=%p grp=%s members=\"%s\"",
+			    ngt, group, val);
+
+		nfound++;
+
+		if ((p = strpbrk(val, "#\n")) != 0) {
+			*p = '\0';
+		}
+		p = val;
+
+		/* Parse val into triples and recursive netgroup references */
+		/*CONSTCOND*/
+		while (1) {
+			ccp			triple[NSS_NETGR_N];
+			int			syntax_err;
+			enum nss_netgr_argn	i;
+
+			while (isspace(*p)) {
+				p++;
+			}
+			if (*p == '\0') {
+				/* Finished processing this particular val */
+				break;
+			}
+			if (*p != '(') {
+				/* Doesn't look like the start of a triple, */
+				/*   so assume it's a recursive netgroup.   */
+				char *start = p;
+				p = strpbrk(start, " \t");
+				if (p == 0) {
+					/* Point p at the final '\0' */
+					p = start + strlen(start);
+				}
+				ngt_insert(ngt, start, (size_t)(p - start));
+				continue;
+			}
+
+			/* Main case:  a (machine, user, domain) triple */
+			p++;
+			syntax_err = 0;
+			for (i = NSS_NETGR_MACHINE; i < NSS_NETGR_N; i++) {
+				char		*start;
+				char		*limit;
+				const char	*terminators = ",) \t";
+
+				if (i == NSS_NETGR_DOMAIN) {
+					/* Don't allow comma */
+					terminators++;
+				}
+				while (isspace(*p)) {
+					p++;
+				}
```

Comment:
You could lose the braces here (not required for single line body)
and in other places like this - then more of this function will appear on my screen